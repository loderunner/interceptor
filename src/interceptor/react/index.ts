export { Provider as InterceptProvider, useIntercepts } from './intercept'
