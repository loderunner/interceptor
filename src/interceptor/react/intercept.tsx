import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import * as intercept from '../intercept'

interface InterceptContextType {
  intercepts: intercept.Intercept[]
  addIntercept: (pattern: string) => void
  removeIntercept: (inter: intercept.Intercept) => void
}

const InterceptContext = createContext<InterceptContextType>({
  intercepts: [],
  addIntercept: () => {},
  removeIntercept: () => {},
})

interface Props {
  children: React.ReactNode
}

export const Provider = ({ children }: Props) => {
  const [intercepts, setIntercepts] = useState([...intercept.intercepts])

  const addIntercept = (pattern: string) => {
    intercept.addIntercept({ pattern, enabled: true })
    setIntercepts([...intercept.intercepts])
  }
  const removeIntercept = (inter: intercept.Intercept) => {
    intercept.removeIntercept(inter)
    setIntercepts([...intercept.intercepts])
  }

  return (
    <InterceptContext.Provider
      value={{ intercepts, addIntercept, removeIntercept }}
    >
      {children}
    </InterceptContext.Provider>
  )
}

export const useIntercepts = () => useContext(InterceptContext)
