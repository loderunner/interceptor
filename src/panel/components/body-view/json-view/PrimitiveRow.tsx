import * as React from 'react'
import { useState } from 'react'

import { Clear as ClearIcon } from '@/icons'

import KeyView from './KeyView'
import PrimitiveView from './PrimitiveView'

import type { JSONPrimitive, JSONValue } from '../JSON'

interface Props {
  k: string
  v: JSONPrimitive
  keyEditable: boolean
  depth: number
  onChangeKey?: (k: string) => void
  onChangeValue?: (v: JSONValue) => void
  onDelete?: () => void
}

const PrimitiveRow = ({
  k,
  v,
  depth,
  onChangeKey,
  onChangeValue,
  onDelete,
  keyEditable = onChangeKey !== undefined,
}: Props) => {
  const [hovering, setHovering] = useState(false)

  return (
    <div
      className="mt-1 flex items-center space-x-1"
      style={{ marginLeft: `${depth}rem` }}
      onMouseEnter={() => setHovering(true)}
      onMouseLeave={() => setHovering(false)}
    >
      <button
        className={`rounded-sm bg-red-500 hover:bg-red-600 active:bg-red-700 ${
          hovering ? '' : ' invisible'
        }`}
      >
        <ClearIcon className="fill-white" onClick={onDelete} />
      </button>
      <KeyView name={k} editable={keyEditable} onChange={onChangeKey} />
      <PrimitiveView value={v} onChange={onChangeValue} />
    </div>
  )
}

export default PrimitiveRow
