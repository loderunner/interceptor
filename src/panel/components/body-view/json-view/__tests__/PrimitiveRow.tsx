import { cleanup, render } from '@testing-library/react'
import * as React from 'react'
import '@testing-library/jest-dom'

import PrimitiveRow from '../PrimitiveRow'

describe('[JSON.PrimitiveRow]', () => {
  afterEach(() => {
    cleanup()
  })

  it('should match initial snapshot', () => {
    const { container } = render(
      <PrimitiveRow depth={0} k="key" v={null} keyEditable />
    )
    expect(container).toMatchSnapshot()
  })
})
