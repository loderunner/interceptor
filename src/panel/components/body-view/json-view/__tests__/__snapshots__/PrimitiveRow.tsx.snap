// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[JSON.PrimitiveRow] should match initial snapshot 1`] = `
<div>
  <div
    class="mt-1 flex items-center space-x-1"
    style="margin-left: 0rem;"
  >
    <button
      class="rounded-sm bg-red-500 hover:bg-red-600 active:bg-red-700  invisible"
    >
      <svg
        class="fill-white"
      />
    </button>
    <pre
      class="text-amber-700"
    >
      key
      :
    </pre>
     
    <div
      class="flex-auto"
    >
      <pre
        class="whitespace-pre-wrap break-all text-purple-500"
      >
        null
      </pre>
    </div>
  </div>
</div>
`;
